pip/commands/show.py

```diff
@@ -12,6 +13,13 @@
 logger = logging.getLogger(__name__)
 
 
+_normalize_re = re.compile(r'[^a-z0-9-.]', re.I)
+
+
+def normalize_name(name):
+    return _normalize_re.sub('-', name.lower())
+
+
```

Comment:
Yes, I've used it in my first [commit](https://github.com/berkerpeksag/pip/commit/6425bc04e86073a4b3b567f4c14fca6897958d3c), but since it's much more strict it broke a couple tests. I'm open to suggestions to handle this with a simpler way.